FirebaseRemoteConfig/SwiftNew/ConfigRealtime.swift

```diff
@@ -0,0 +1,665 @@
+        completion(Data())
+      }
+    }
+  }
+
+  // MARK: - Retry Helpers
+
+  func canMakeConnection() -> Bool {
+    let noRunningConnection = dataTask == nil || dataTask?.state != .running
+    return noRunningConnection && listeners.count == 0 && !isInBackground && !isRealtimeDisabled
+  }
+
+  func retryHTTPConnection() {
+    realtimeLockQueue.async {
+      guard self.remainingRetryCount > 0 else {
+        let error = NSError(domain: RemoteConfigUpdateErrorDomain,
+                            code: RemoteConfigUpdateError.streamError.rawValue,
+                            userInfo: [
+                              NSLocalizedDescriptionKey: "Unable to connect to the server. Check your connection and try again.",
+                            ])
+        RCLog.error("I-RCN000014", "Cannot establish connection. Error: \(error)")
+        self.propagateErrors(error)
+        return
+      }
+      if self.canMakeConnection() {
+        self.remainingRetryCount -= 1
+        self.settings.realtimeRetryCount += 1
+        let backoffInterval = self.settings.realtimeBackoffInterval()
+        self.realtimeLockQueue.asyncAfter(deadline: .now() + backoffInterval) {
+          self.beginRealtimeStream()
+        }
+      }
+    }
+  }
+
+  private func backgroundChangeListener() {
+    #if canImport(UIKit)
+      NotificationCenter.default.addObserver(self,
+                                             selector: #selector(willEnterForeground),
+                                             name: UIApplication
+                                               .willEnterForegroundNotification,
+                                             object: nil)
+      NotificationCenter.default.addObserver(self,
+                                             selector: #selector(didEnterBackground),
+                                             name: UIApplication.didEnterBackgroundNotification,
+                                             object: nil)
+    #elseif canImport(AppKit)
+      NotificationCenter.default.addObserver(self,
+                                             selector: #selector(willEnterForeground),
+                                             name: NSApplication.willBecomeActiveNotification,
+                                             object: nil)
+      NotificationCenter.default.addObserver(self,
+                                             selector: #selector(didEnterBackground),
+                                             name: NSApplication.didResignActiveNotification,
+                                             object: nil)
+    #endif
+  }
+
+  @objc private func willEnterForeground() {
+    realtimeLockQueue.async {
+      self.isInBackground = false
+      self.beginRealtimeStream()
+    }
+  }
+
+  @objc private func didEnterBackground() {
+    realtimeLockQueue.async {
+      self.isInBackground = true
+      self.pauseRealtimeStream()
```

Comment:
not sure if it makes a difference but the ordering was reversed in objc 
```suggestion
      self.pauseRealtimeStream()
      self.isInBackground = true
```

Reply:
maybe? Anyway, no reason not to keep in the same. changed.